x/streamer/keeper/hooks.go

```diff
@@ -34,54 +36,118 @@ func (k Keeper) Hooks() Hooks {
 /* -------------------------------------------------------------------------- */
 
 // BeforeEpochStart is the epoch start hook.
-func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string) error {
+	// Move upcoming streams to active if start time reached
+	upcomingStreams := k.GetUpcomingStreams(ctx)
+	for _, s := range upcomingStreams {
+		if !ctx.BlockTime().Before(s.StartTime) {
+			err := k.moveUpcomingStreamToActiveStream(ctx, s)
+			if err != nil {
+				return fmt.Errorf("move upcoming stream to active stream: %w", err)
+			}
+		}
+	}
+
+	var toStart []types.Stream
+
+	activeStreams := k.GetActiveStreams(ctx)
+	for _, s := range activeStreams {
+		if epochIdentifier == s.DistrEpochIdentifier {
+			toStart = append(toStart, s)
+		}
+	}
+
+	// Update streams with respect to a new epoch and save them
+	for _, s := range toStart {
+		updated, err := k.UpdateStreamAtEpochStart(ctx, s)
+		if err != nil {
+			return fmt.Errorf("update stream '%d' at epoch start: %w", s.Id, err)
+		}
+		// Save the stream
+		err = k.SetStream(ctx, &updated)
+		if err != nil {
+			return fmt.Errorf("set stream: %w", err)
+		}
+	}
+
+	err := ctx.EventManager().EmitTypedEvent(&types.EventEpochStart{
+		ActiveStreamsNum: uint64(len(toStart)),
+	})
+	if err != nil {
+		return fmt.Errorf("emit typed event: %w", err)
+	}
+
 	return nil
 }
 
 // AfterEpochEnd is the epoch end hook.
-func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
-	streams := k.GetUpcomingStreams(ctx)
-	// move to active if start time reached
-	for _, stream := range streams {
-		if !ctx.BlockTime().Before(stream.StartTime) {
-			if err := k.moveUpcomingStreamToActiveStream(ctx, stream); err != nil {
-				return err
-			}
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string) (sdk.Coins, error) {
+	activeStreams := k.GetActiveStreams(ctx)
+
+	var toDistribute []types.Stream
+
+	for _, s := range activeStreams {
+		if epochIdentifier == s.DistrEpochIdentifier {
+			toDistribute = append(toDistribute, s)
 		}
 	}
```

Comment:
lets wrap it in `k.GetActiveStreamsForEpoch(ctx, epochIdentifier)`, as it's used above as well